pkg/prometheus/statefulset.go

```diff
@@ -266,7 +266,12 @@ func makeStatefulSetService(p *monitoringv1.Prometheus, config Config) *v1.Servi
 func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config, ruleConfigMapNames []string) (*appsv1.StatefulSetSpec, error) {
 	// Prometheus may take quite long to shut down to checkpoint existing data.
 	// Allow up to 10 minutes for clean termination.
-	terminationGracePeriod := int64(600)
+	var terminationGracePeriod int64
+	if nil != p.Spec.TerminationGracePeriodSeconds {
+		terminationGracePeriod = *p.Spec.TerminationGracePeriodSeconds
+	} else {
+		terminationGracePeriod = int64(300)
```

Comment:
why changing the default?

Reply:
I modify the v0.23.0,not the master

Reply:
v0.23.0 is a very old release that isn't supported anymore. Please rebase on top of the `master` branch.